tests/foreman/destructive/test_ping.py

```diff
@@ -21,23 +21,44 @@
 pytestmark = pytest.mark.destructive
 
 
-def test_negative_ping_fail_status_code(target_sat):
-    """Negative test to verify non-zero status code of ping fail
+@pytest.fixture(scope='module')
+def tomcat_service_teardown(request, module_target_sat):
+    assert module_target_sat.cli.Service.stop(options={'only': 'tomcat.service'}).status == 0
+
+    @request.addfinalizer
+    def _finalize():
+        assert module_target_sat.cli.Service.start(options={'only': 'tomcat.service'}).status == 0
+
+    yield module_target_sat
```

Comment:
no need to yield satellite here, you can just yield and use `module_target_sat` in the test function.